src/common_i2c.cpp

```diff
@@ -290,6 +311,33 @@ int i2cDev::i2c_transfer(uint8_t tx_len, uint8_t* tx_data, uint8_t rx_len,
     return res;
 }
 
+bool i2cDev::isSpamingToLog(int res, std::stringstream& ss)
+{
+    I2cContext& context = getI2cContext(deviceLabel, i2cAddr);
+    auto& numLogErrors = context.numLogErrors;
+
+    const uint32_t maxLogErrors = 3;
+
+    if (res > 0)
+    {
+        numLogErrors = 0;
+        return false;
+    }
+
+    if (numLogErrors == maxLogErrors)
+    {
+        ss << "... (Detected multiple errors. Stoping spam to log.)";
+        log<level::ERR>(ss.str().c_str());
+    }
+
+    if (numLogErrors < std::numeric_limits<uint32_t>::max())
```

Comment:
Looks weird.
Likely it would be more readable:
```cpp
if (numLogErrors >= maxLogErrors)
{
        ss << "... (Detected multiple errors. Stoping spam to log.)";
        log<level::ERR>(ss.str().c_str());
}
else
{
        bymLegErrors++;
}
```

Reply:
reworked